src/Mesh/DerivedData.C

```diff
@@ -802,6 +804,86 @@ genEsued( const std::vector< std::size_t >& inpoel,
   return std::make_pair( std::move(esued1), std::move(esued2) );
 }
 
+std::size_t
+genNbfacTet( std::size_t tnbfac,
+             const std::vector< std::size_t >& inpoel,
+             const std::vector< std::size_t >& t_triinpoel,
+             std::vector< std::size_t >& triinpoel )
+// *****************************************************************************
+//  Generate the number of boundary-faces on this chare/worker/mesh-partition.
```

Comment:
I would try to phrase this (brief function title) in a more complete and more generic fashion, something like: _"Generate the number of faces and the triangle face connectivity for a chunk of a full mesh"_.

Then in `\detail`, it would be great to have a specific description on what is being done here. I'll attempt and you correct if I'm wrong: _"This function takes a mesh by its domain-element (tetrahedron-connectivity) in inpoel and a boundary-face (triangle) connectivity in t_triinpoel. Based on these two arrays, it searches for those faces of t_triinpoel that are also in inpoel and as a result it generates (1) the number of boundary faces shared with the mesh in inpoel and (2) the intersection of the triangle element connectivity whose faces are shared with inpoel. An example use case is where t_triinpoel contains the connectivity for the boundary of the full problem/mesh and inpoel contains the connectivity for only a chunk of an already partitioned mesh. This function then intersects t_triinpoel with inpoel and returns only those faces that share nodes with inpoel."_

Also, would it be better to name `t_triinpoel` as `triinpoel_complete` or `all_triinpoel`, something like that? And/or name `triinpoel` as `sub_triinpoel`?

Reply:
I think it's better to leave triinpoel as it is, since all other connectivity arrays don't have a "_sub". However, I'll rename t_triinpoel as triinpoel_complete. Is that okay?

Reply:
Sounds good.